vms/organization/views.py

```diff
@@ -2,105 +2,80 @@
-                )
-    else:
-        return render(request, 'vms/no_admin_rights.html')
-
-
-@login_required
-def delete(request, organization_id):
-    if is_admin(request):
-        if request.method == 'POST':
-            result = delete_organization(organization_id)
-            if result:
-                return HttpResponseRedirect(reverse('organization:list'))
-            else:
-                return render(request, 'organization/delete_error.html')
-        return render(
-            request,
-            'organization/delete.html',
-            {'organization_id': organization_id}
-            )
-    else:
-        return render(request, 'vms/no_admin_rights.html')
-
-
-@login_required
-def edit(request, organization_id):
-    if is_admin(request):
-        organization = None
-        if organization_id:
-            organization = get_organization_by_id(organization_id)
-
-        if request.method == 'POST':
-            form = OrganizationForm(request.POST, instance=organization)
-            if form.is_valid():
-                form.save()
-                return HttpResponseRedirect(reverse('organization:list'))
-            else:
-                return render(
-                    request,
-                    'organization/edit.html',
-                    {'form': form, }
-                    )
+            return super(AdministratorLoginRequiredMixin, self).dispatch(request, *args, **kwargs)
+
+
+class OrganizationCreateView(LoginRequiredMixin, AdministratorLoginRequiredMixin, FormView):
+    template_name = 'organization/create.html'
+    form_class = OrganizationForm
+
+    def form_valid(self, form):
+        form.save()
+        return HttpResponseRedirect(reverse('organization:list'))
+
+
+class OrganizationDeleteView(LoginRequiredMixin, AdministratorLoginRequiredMixin, DeleteView):
+    template_name = 'organization/delete.html'
+    success_url = reverse_lazy('organization:list')
+    model = Organization
+
+    def get_object(self, queryset=None):
+        organization_id = self.kwargs['organization_id']
+        org = Organization.objects.get(pk=organization_id)
+        if org:
+            return org
+
+    def delete(self, request, *args, **kwargs):
+        org = self.get_object()
+        volunteers_in_organization = org.volunteer_set.all()
+        administrators_in_organization = org.administrator_set.all()
+        if org and (not volunteers_in_organization) and (not administrators_in_organization):
+            org.delete()
+            return HttpResponseRedirect(self.success_url)
         else:
-            form = OrganizationForm(instance=organization)
-            return render(
-                request,
-                'organization/edit.html',
-                {'form': form, }
-                )
-    else:
-        return render(request, 'vms/no_admin_rights.html')
-
-
-@login_required
-def list(request):
-    organization_list = get_organizations_ordered_by_name()
```

Comment:
@amruthasangeeth Please do this change and update the pull request or create a new one.